core/src/test/scala/unit/kafka/server/KafkaApisTest.scala

```diff
@@ -5289,6 +5304,156 @@ class KafkaApisTest extends Logging {
+
+    assertEquals(Errors.INVALID_RECORD_STATE.code(), responseData.errorCode())
+    assertEquals(1, topicResponses.size())
+    assertEquals(1, topicResponses.get(0).partitions().size())
+    assertEquals(Errors.NONE.code(), topicResponses.get(0).partitions().get(0).acknowledgeErrorCode())
+  }
+
+  @Test
+  def testHandleShareFetchRequestFinalFetchWithErroneousCachedPartitionsData(): Unit = {
+    val topicName = "foo"
+    val topicId = Uuid.randomUuid()
+    addTopicToMetadataCache(topicName, 1, topicId = topicId)
+    val memberId: Uuid = Uuid.ZERO_UUID
+
+    when(sharePartitionManager.acknowledgeShareSessionCacheUpdate(
+      any(),
+      any(),
+      any(),
+      any()
+    )).thenReturn(Errors.NONE)
+    when(sharePartitionManager.newContext(any(), any(), any(), any(), any())).thenReturn(
+      new FinalContext(new mutable.HashMap[TopicIdPartition, ShareFetchRequest.SharePartitionData]().asJava)
+    )
+
+    when(clientQuotaManager.maybeRecordAndGetThrottleTimeMs(
+      any[RequestChannel.Request](), anyDouble, anyLong)).thenReturn(0)
+
+    when(sharePartitionManager.acknowledge(
+      anyString(),
+      anyString(),
+      any()
+    )).thenReturn(CompletableFuture.completedFuture(Map[TopicIdPartition, ShareAcknowledgeResponseData.PartitionData](
+      new TopicIdPartition(topicId, new TopicPartition(topicName, 0)) ->
+        new ShareAcknowledgeResponseData.PartitionData()
+          .setPartitionIndex(0)
+          .setErrorCode(Errors.NONE.code())
+    ).asJava))
+
+    when(sharePartitionManager.cachedTopicIdPartitionsInShareSession(
+      anyString(),
+      any()
+    )).thenThrow(new ShareSessionNotFoundException("share session does not exist"))
+
+    val shareFetchRequestData = new ShareFetchRequestData().
+      setGroupId("group").
+      setMemberId(memberId.toString).
+      setShareSessionEpoch(ShareFetchMetadata.FINAL_EPOCH).
+      setTopics(List(new ShareFetchRequestData.FetchTopic().
+        setTopicId(topicId).
+        setPartitions(List(
+          new ShareFetchRequestData.FetchPartition()
+            .setPartitionIndex(0)
+            .setPartitionMaxBytes(0)
+            .setAcknowledgementBatches(List(
+              new ShareFetchRequestData.AcknowledgementBatch()
+                .setBaseOffset(11)
+                .setLastOffset(20)
+            ).asJava)
+        ).asJava)
+      ).asJava)
+
+    val shareFetchRequest = new ShareFetchRequest.Builder(shareFetchRequestData).build(ApiKeys.SHARE_FETCH.latestVersion)
+    val request = buildRequest(shareFetchRequest)
+    kafkaApis = createKafkaApis(overrideProperties = Map(KafkaConfig.ShareGroupEnableProp -> "true"))
+    kafkaApis.handleShareFetchRequest(request)
+    val response = verifyNoThrottling[ShareFetchResponse](request)
+    val responseData = response.data()
+    val topicResponses = response.data().responses()
+
+    assertEquals(Errors.SHARE_SESSION_NOT_FOUND.code(), responseData.errorCode())
```

Comment:
I think this error code means that the combination of group ID and member ID wasn't in the share session cache. This request looks like it's closing the share session and including some acknowledgements that should be actioned before the final round of releasing any others for this session. So, if the share session does exist, I expect the acks to be performed, and then release, and then return NONE. Maybe I don't quite understand what this test is doing.

Reply:
yeah, I agree, if there is a case of SHARE_SESSION_NOT_FOUND, it will be caught while doing he acks so we don't even need this test. I am modifying it to make sense.